src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1625,6 +1775,12 @@ Node TheoryStringsRewriter::rewriteContains( Node node ) {
     Node ret = NodeManager::currentNM()->mkConst(false);
     return returnRewrite(node, ret, "ctn-len-ineq");
   }
+  else if (checkEntailArith(len_n2, len_n1, false))
```

Comment:
This should be subsumed by #2512.

Reply:
I would argue for keeping this rewrite. This rewrite + generalizing inferEqsFromContains to apply bidirectionality to equalities should subsume the current.